### DISCUSSION TITLE
(토론) 레시피 스텝 제작 할 때, 생각하지 못한 경우의 수

### DISCUSSION THREAD

**Discussion by Hogu59** [woowacourse-teams/2024-pengcook#156] [General]
현재 매 스텝마다 저장합니다.
case1)
1. 사용자가 레시피 제작 10단계까지 만들었습니다.
2. 11단계까지 만들다가 그냥 앱을 껐습니다.
3. 그리고 레시피 제작을 포기하고 안 만들겠다고 생각합니다.
-> 서버의 데이터는 가비지로 남겨져 있는건가요?

case2)
1. 사용자가 레시피를 제작하다가 나갔습니다.
2. 다시 제작하려고 레시피개요 제작 화면으로 돌아와서 다시 적습니다.
3. 여기서 제작하기를 눌렀을때 받아오는 recipeId 값은 아마 다를텐데 어떻게 기존에 적던 레시피로 접근이 가능할까요..?

**Comment by oshyun00** [comment]
매 스텝마다 저장하는 방식으로 변경했는데, 이 경우 `레시피 상태`를 추가로 관리해야할 것 같아요.
(현재 구현되어있는 로직은 구분이 없어 레시피 조회 시 작성중인 레시피 정보들도 리턴됨)

(1) 레시피 생성 시 `작성중` 상태로 저장
(2) `작성완료` 버튼을 누르면 `완료` 상태로 변경
(3) 레시피 조회할 때에는 `완료` 상태의 레시피만 조회

case1 - 3)
이 부분은 정책으로 관리해야할 것 같습니다.
최초 작성 이후 일정 기간이 지나면 DB에서 삭제하거나, 작성취소 기능을 추가할 수 있습니다.

case2 - 2)
레시피 개요로 돌아와서 `다시 적게`되면 `새로운 레시피를 생성`하는 것이므로 새로운 레시피ID를 받는게 자연스러운 것 같습니다.

case2-3)
레시피 상태를 관리하게 되면 `본인이 작성중인 레시피`를 확인할 수 있습니다.
그렇게 되면 해당 레시피 ID로 정보를 호출해서 작성중인 내역을 조회할 수 있습니다.
-> 이 부분은 기능 추가 측면인 것 같아서 좀 더 논의가 필요할 것 같습니다.

<참고>
(1) 블로그에서 `글쓰기`를 눌렀을 때, 이전에 작성중인 내역이 있으면 `이전에 작성중인 내용이 있습니다. 불러오시겠습니까?`라고 물어보고, 만약 거절하면 이전에 남아있던 내역은 삭제되어 다시 불러올 수 없음
(2) 블로그 '임시저장' 기능을 활용해 임시로 저장해놨다가 작성중인 내용을 불러올 수 있는데, 일정 기간이 지나면 삭제됨
<img width="393" alt="스크린샷 2024-08-03 오후 4 37 59" src="https://github.com/user-attachments/assets/bba2e3f7-fdd5-4ad6-9451-1627842966c3">

**Reply by Hogu59** [reply]
흠 그러면 상태가 추가되면 전부 처리가 가능할수 있겠네요. 그러면 도메인로직으로 생각해볼때, recipe 와 recipeStep 모두 상태를 가지고 있어야하는 걸까요?
일단 추가적인 논의가 좀 필요하긴 하겠네요. 일단은 pr주기를 짧게 가져가기 위해 이상적인 상황으로 작성을 하고, 리팩토링으로 변경하는 것도 괜찮겠다는 생각이 드네요.
